h2o-algos/src/main/java/hex/DataInfo.java

```diff
@@ -967,13 +967,16 @@ public final int getCategoricalId(int cid, int val) {
     boolean isIWV = isInteractionVec(cid);
     if(val == -1) { // NA
       val = _catNAFill[cid];
-    } else if( !_useAllFactorLevels && !isIWV )  // categorical interaction vecs drop reference level in a special way
-      val -= 1;
+    }
+
+    if (!_useAllFactorLevels && !isIWV) {  // categorical interaction vecs drop reference level in a special way
+      val = val - 1;
+    }
     if(val < 0) return -1; // column si to be skipped
     int [] offs = fullCatOffsets();
     int expandedVal = val + offs[cid];
     if(expandedVal >= offs[cid+1]) {  // previously unseen level
-      assert _valid:"Categorical value out of bounds, got " + val + ", next cat starts at " + fullCatOffsets()[cid+1];
+      assert _valid : "Categorical value out of bounds, got " + val + ", next cat starts at " + fullCatOffsets()[cid + 1];
       if(_skipMissing)
         return -1;
       val = _catNAFill[cid];
```

Comment:
This is also not correct. Anytime you use _catNAFill, you need to consider the adjustment of indices.

    if (!_useAllFactorLevels && !isIWV) {  // categorical interaction vecs drop reference level in a special way
      val = val - 1;
    }

Reply:
This one has been fixed.

Reply:
ARE YOU SURE? :P

Reply:
Yes ! 💃